smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/lang/DocumentationPreprocessor.kt

```diff
@@ -200,12 +200,19 @@ private fun Node.hasAncestor(predicate: (Node) -> Boolean): Boolean =
 private fun Node.isList() =
     nodeName().let { it == "ul" || it == "ol" }
 
-private fun TextNode.markdownText() =
-    text()
+private fun Node.isPreformat() =
+    nodeName().let { it == "code" || it == "pre" }
+
+private fun TextNode.markdownText(): String {
+    // If we're inside a preformat block, everything is literal, ie. no escapes required.
+    if (hasAncestor(Node::isPreformat)) return text()
+
+    return text()
         // Replace square brackets with escaped equivalents so that they are not rendered as invalid Markdown
         // links.
         .replace("[", "&#91;")
         .replace("]", "&#93;")
+}
```

Comment:
**Style**: Multiple returns? Gross! Could be simplified with a `when` block:

```kotlin
private fun TextNode.markdownText(): String = when {
    hasAncestor(Node::isPreformat)) -> text()

    // Replace square brackets with escaped equivalents so that they are not rendered as invalid Markdown
    // links.
    else -> text().replace("[", "&#91;").replace("]", "&#93;")
}
```